src/components/composite/aspect/AspectForm/AspectForm.tsx

```diff
@@ -47,10 +57,19 @@ const AspectForm: FC<AspectFormProps> = ({ aspect, templateDescription, edit = f
     nameID: aspect?.nameID ?? '',
     description: aspect?.description ?? templateDescription ?? '',
     tagsets,
+    aspects: aspects && aspects.length > 0 ? aspects : [],
   };
 
   const validationSchema = yup.object().shape({
-    name: nameSegmentSchema.fields?.name || yup.string(),
+    name: yup
+      .string()
+      .test('is-valid-name', '${path} is already used in another aspect', value => {
+        if (value && aspects && !aspects.includes(value)) return true;
+        return false;
+      })
+      .required(i18next.t('forms.validations.required'))
+      .min(3, 'Name should be at least 3 symbols long')
+      .max(128, 'Exceeded the limit of 128 characters'),
```

Comment:
These can also be moved to the translation file; also is something wrong with yup's default messages?